src/Microsoft.ML.AutoML/API/AutoCatalog.cs

```diff
@@ -13,6 +14,11 @@ public sealed class AutoCatalog
     {
         private readonly MLContext _context;
 
+        /// <summary>
+        /// save some intermediate value
+        /// </summary>
+        public static Dictionary<string, object> ValuePairs { get; set; } = new Dictionary<string, object>();
```

Comment:
What is this for? This doesn't seem like a good API, we should stay away from static variables in our API.

Reply:
That's the shortcut to make it functionally working... Because `MatrixFactorizationTrainer` accepts both `MatrixColumnIndexColumnName` and `MatrixRowIndexColumnName` as input feature columns, but currently `AutoML` requires it's trainers only accept one `feature` columns as input. To change that we need to do some refactor work on `AutoML` side, so that's just the quick way to make it work

Reply:
See my feedback about creating two explicit ColumnPurposes for these two.